gwpy/cli/cliproduct.py

```diff
@@ -469,21 +469,29 @@ def _filter_timeseries(data, highpass=None, lowpass=None, notch=None):
         if all(x is None for x in (highpass, lowpass, notch)):
             return data
 
-        # build ZPK
-        zpks = []
+        # apply sos filters one at a time to avoid problems combining them
+
+        sos = None
         if highpass is not None and lowpass is not None:
-            zpks.append(filter_design.bandpass(highpass, lowpass,
-                                               data.sample_rate))
+            sos = filter_design.bandpass(highpass, lowpass, data.sample_rate,
+                                         ftype='butter', output='sos')
         elif highpass is not None:
-            zpks.append(filter_design.highpass(highpass, data.sample_rate))
+            sos = filter_design.highpass(highpass, data.sample_rate,
+                                         ftype='butter', output='sos')
         elif lowpass is not None:
-            zpks.append(filter_design.lowpass(lowpass, data.sample_rate))
+            sos = filter_design.lowpass(lowpass, data.sample_rate,
+                                        ftype='butter', output='sos')
+
+        if sos is not None:
+            ret = data.filter(sos)
+        elif notch:
+            ret = data
```

Comment:
Why not just overload the local `data` variable name?
```suggestion
        if sos is not None:
            data = data.filter(sos)
```

Reply:
ok